Compilation failure

Hi,

`onyx` and `spider-gazelle` are facing **compilation issue** `here`

Using `jrei/crystal-alpine:latest` I have

~~~crystal
Error target server failed to compile:
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/lib/libxml2.a(xzlib.o): in function `xz_decomp':
/home/buildozer/aports/main/libxml2/src/libxml2-2.9.10/xzlib.c:581: undefined reference to `lzma_code'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/lib/libxml2.a(xzlib.o): in function `xz_head':
/home/buildozer/aports/main/libxml2/src/libxml2-2.9.10/xzlib.c:411: undefined reference to `lzma_auto_decoder'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/lib/libxml2.a(xzlib.o): in function `is_format_lzma':
/home/buildozer/aports/main/libxml2/src/libxml2-2.9.10/xzlib.c:312: undefined reference to `lzma_properties_decode'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/lib/libxml2.a(xzlib.o): in function `__libxml2_xzclose':
/home/buildozer/aports/main/libxml2/src/libxml2-2.9.10/xzlib.c:799: undefined reference to `lzma_end'
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o '/usr/src/app/bin/server'  -rdynamic -static  /usr/lib/libxml2.a -lz `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libssl || printf %s '-lssl -lcrypto'` `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libcrypto || printf %s '-lcrypto'` /usr/lib/libpcre.a /usr/lib/libm.a /usr/lib/libgc.a /usr/lib/libpthread.a /usr/lib/crystal/core/ext/libcrystal.a /usr/lib/libevent.a /usr/lib/librt.a -L/usr/lib -L/usr/local/lib`
~~~

when trying to build statically
~~~
shards build --production --release --no-debug --static
~~~

Unfortunately @j8r, your [patch](https://github.com/j8r/dockerfiles/commit/6561dc2db61bcd233e648dd6b358ad9a44e50904) does not fix it

Regards,

/cc @vladfaust @stakach

--- a/crystal/Dockerfile
+++ b/crystal/Dockerfile
@@ -5,7 +5,7 @@ WORKDIR /usr/src/app
 COPY . ./
 
 RUN shards install
-RUN shards build --production --release --no-debug --static
+RUN shards build --production --release --no-debug --static {{#build_opts}} {{{.}}} {{/build_opts}}
 
 FROM alpine
 

--- a/crystal/onyx/config.yaml
+++ b/crystal/onyx/config.yaml
@@ -1,3 +1,5 @@
 framework:
   website: onyxframework.org
   version: 0.5
+
+build_opts: '--link-flags "-lxml2 -llzma"'

--- a/tools/src/make.cr
+++ b/tools/src/make.cr
@@ -75,6 +75,9 @@ class App < Admiral::Command
               if framework_config.as_h.has_key?("image")
                 params["image"] = framework_config.as_h["image"].to_s
               end
+              if framework_config.as_h.has_key?("build_opts")
+                params["build_opts"] = framework_config.as_h["build_opts"].to_s
+              end
               if framework_config.as_h.has_key?("deps")
                 deps = [] of String
                 framework_config["deps"].as_a.each do |dep|

